connectors/sources/google_cloud_storage.py

```diff
@@ -199,17 +148,114 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+
+class GoogleCloudStorageDataSource(BaseDataSource):
+    """Google Cloud Storage"""
+
+    name = "Google Cloud Storage"
+    service_type = "google_cloud_storage"
+
+    def __init__(self, configuration):
+        """Set up the connection to the Google Cloud Storage Client.
+
+        Args:
+            configuration (DataSourceConfiguration): Object of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.enable_content_extraction = self.configuration["enable_content_extraction"]
+
+        self.google_storage_client = None
```

Comment:
```suggestion
        self._google_storage_client = None
```

I'd make it private to indicate that it should not be used.